File: frontend/src/types/project.ts
import type { Episode } from './episode'

/** 项目状态 */
export type ProjectStatus = 'draft' | 'parsing' | 'parsed' | 'generating' | 'composing' | 'completed' | 'failed'

export interface WorkflowDefaults {
  video_provider_key: string | null
  tts_provider_key: string | null
  lipsync_provider_key: string | null
  provider_payload_defaults: {
    video: Record<string, unknown>
    tts: Record<string, unknown>
    lipsync: Record<string, unknown>
  }
}

/** 项目 */
export interface Project {
  id: string
  name: string
  description: string | null
  script_text: string | null
  status: ProjectStatus
  episode_count: number
  panel_count: number
  generated_panel_count: number
  character_count: number
  workflow_defaults: WorkflowDefaults
  created_at: string
  updated_at: string
}

/** 项目列表项 */
export interface ProjectListItem {
  id: string
  name: string
  description: string | null
  status: ProjectStatus
  episode_count: number
  panel_count: number
  generated_panel_count: number
  character_count: number
  created_at: string
  updated_at: string
}

/** 创建项目请求 */
export interface ProjectCreate {
  name: string
  description?: string
  workflow_defaults?: Partial<WorkflowDefaults>
}

/** 更新项目请求 */
export interface ProjectUpdate {
  name?: string
  description?: string
  script_text?: string
  workflow_defaults?: Partial<WorkflowDefaults> | null
}

export interface ProjectWorkspaceResourceSummary {
  character_entity_count: number
  bound_character_entity_count: number
  location_entity_count: number
  bound_location_entity_count: number
  voice_asset_count: number
  panel_count: number
  clip_count: number
  ready_clip_count: number
  failed_clip_count: number
  composition_count: number
}

export interface ProjectWorkspacePreview {
  id: string
  status: string
  duration_seconds: number
  created_at: string | null
  updated_at: string | null
}

export interface ProjectWorkspace {
  project: Project
  episodes: Episode[]
  resource_summary: ProjectWorkspaceResourceSummary
  latest_preview: ProjectWorkspacePreview | null
  recommended_episode_id: string | null
  recommended_step: string
}

/** 状态标签颜色映射 */
export const STATUS_MAP: Record<ProjectStatus, { label: string; color: string }> = {
  draft: { label: '草稿', color: 'default' },
  parsing: { label: '解析中', color: 'processing' },
  parsed: { label: '已解析', color: 'cyan' },
  generating: { label: '生成中', color: 'processing' },
  composing: { label: '合成中', color: 'processing' },
  completed: { label: '已完成', color: 'success' },
  failed: { label: '失败', color: 'error' },
}

File: backend/app/schemas/project.py
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field

from app.schemas.episode import EpisodeResponse


class WorkflowDefaultsPayload(BaseModel):
    """项目级默认工作流配置。"""

    video_provider_key: str | None = Field(default=None, min_length=1, max_length=120)
    tts_provider_key: str | None = Field(default=None, min_length=1, max_length=120)
    lipsync_provider_key: str | None = Field(default=None, min_length=1, max_length=120)
    provider_payload_defaults: dict[str, dict[str, Any]] = Field(default_factory=dict)


class ProjectCreate(BaseModel):
    """创建项目请求"""

    name: str = Field(..., min_length=1, max_length=200)
    description: str | None = None
    workflow_defaults: WorkflowDefaultsPayload | None = None


class ProjectUpdate(BaseModel):
    """更新项目请求"""

    name: str | None = Field(None, min_length=1, max_length=200)
    description: str | None = None
    script_text: str | None = None
    workflow_defaults: WorkflowDefaultsPayload | None = None


class ProjectScriptWorkspaceEpisodePayload(BaseModel):
    id: str | None = None
    title: str | None = Field(default=None, max_length=200)
    summary: str | None = None
    script_text: str | None = None
    video_provider_key: str | None = Field(default=None, min_length=1, max_length=120)
    tts_provider_key: str | None = Field(default=None, min_length=1, max_length=120)
    lipsync_provider_key: str | None = Field(default=None, min_length=1, max_length=120)
    provider_payload_defaults: dict[str, dict[str, Any]] = Field(default_factory=dict)
    skipped_checks: list[str] = Field(default_factory=list)


class ProjectScriptWorkspaceUpdate(BaseModel):
    script_text: str = Field(min_length=1)
    workflow_defaults: WorkflowDefaultsPayload
    episodes: list[ProjectScriptWorkspaceEpisodePayload] = Field(min_length=1)


class ProjectResponse(BaseModel):
    """项目响应"""

    id: str
    name: str
    description: str | None
    script_text: str | None
    status: str
    episode_count: int = 0
    panel_count: int = 0
    generated_panel_count: int = 0
    character_count: int = 0
    workflow_defaults: WorkflowDefaultsPayload
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ProjectListItem(BaseModel):
    """项目列表项（不含剧本全文）"""

    id: str
    name: str
    description: str | None
    status: str
    episode_count: int = 0
    panel_count: int = 0
    generated_panel_count: int = 0
    character_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ProjectWorkspaceResourceSummary(BaseModel):
    character_entity_count: int = 0
    bound_character_entity_count: int = 0
    location_entity_count: int = 0
    bound_location_entity_count: int = 0
    voice_asset_count: int = 0
    panel_count: int = 0
    clip_count: int = 0
    ready_clip_count: int = 0
    failed_clip_count: int = 0
    composition_count: int = 0


class ProjectWorkspacePreview(BaseModel):
    id: str
    status: str
    duration_seconds: float = 0
    created_at: str | None = None
    updated_at: str | None = None


class ProjectWorkspaceResponse(BaseModel):
    project: ProjectResponse
    episodes: list[EpisodeResponse]
    resource_summary: ProjectWorkspaceResourceSummary
    latest_preview: ProjectWorkspacePreview | None = None
    recommended_episode_id: str | None = None
    recommended_step: str = "script"
